internal/secrets/secrets.go

```diff
@@ -0,0 +1,165 @@
+// Package secrets provides functions for retrieving local and remote secrets
```

Comment:
Should the docs mention how this supports injecting/interpolating the secret values?